src/commands/cloud-run/flare.ts

```diff
@@ -114,21 +130,60 @@ export class CloudRunFlareCommand extends Command {
     const configStr = util.inspect(config, false, 10, true)
     this.context.stdout.write(`\n${configStr}\n`)
 
-    // Save and zip service configuration
-    this.context.stdout.write(chalk.bold('\n💾 Saving configuration...\n'))
-    const rootFolderPath = path.join(process.cwd(), FLARE_OUTPUT_DIRECTORY)
+    // Get logs
+    const logFileMappings = new Map<CloudRunLog[], string>()
+    if (this.withLogs) {
+      this.context.stdout.write(chalk.bold('\n📖 Getting logs...\n'))
+
+      const logsConfig: LogConfig[] = [
+        {type: 'total', fileName: ALL_LOGS_FILE_NAME},
+        {type: 'warning', severityFilter: ' AND severity>="WARNING"', fileName: WARNING_LOGS_FILE_NAME},
+        {type: 'error', severityFilter: ' AND severity>="ERROR"', fileName: ERRORS_LOGS_FILE_NAME},
+        {type: 'debug', severityFilter: ' AND severity="DEBUG"', fileName: DEBUG_LOGS_FILE_NAME},
+      ]
+
+      const logClient = new Logging({projectId: this.project})
+      for (const logConfig of logsConfig) {
+        try {
+          const logs = await getLogs(logClient, this.service!, this.region!, logConfig.severityFilter)
+          if (logs.length === 0) {
+            this.context.stdout.write(`• No ${logConfig.type} logs were found\n`)
+          } else {
+            this.context.stdout.write(`• Found ${logs.length} ${logConfig.type} logs\n`)
+            logFileMappings.set(logs, logConfig.fileName)
+          }
+        } catch (err) {
+          const msg = err instanceof Error ? err.message : ''
+          this.context.stderr.write(`• Unable to get ${logConfig.type} logs: ${msg}\n`)
+        }
+      }
+    }
+
     try {
-      // Delete folder if it already exists
+      // Create folders
+      const rootFolderPath = path.join(process.cwd(), FLARE_OUTPUT_DIRECTORY)
+      const logsFolderPath = path.join(rootFolderPath, LOGS_DIRECTORY)
+      this.context.stdout.write(chalk.bold(`\n💾 Saving files to ${rootFolderPath}...\n`))
       if (fs.existsSync(rootFolderPath)) {
         deleteFolder(rootFolderPath)
       }
+      const subFolders = []
+      if (logFileMappings.size > 0) {
+        subFolders.push(logsFolderPath)
+      }
```

Comment:
I use this boolean to determine whether or not to create a logs folder. There is an edge case where `--with-logs` is included, but for whatever reason no logs are found (maybe the service hasn't been ran in 24 hours). In this case, it's unnecessary to create a logs folder.

Reply:
What about telling the user that since no logs were found, we are not creating it? I guess this will only apply if they have `--with-logs` set

Reply:
NVM, this is handled in "no logs were found".